crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -135,613 +175,680 @@ pub fn walk_stmt<'a, V>(visitor: &mut V, stmt: &'a Stmt)
 where
     V: PreorderVisitor<'a> + ?Sized,
 {
-    match stmt {
-        Stmt::Expr(ast::StmtExpr { value, range: _ }) => visitor.visit_expr(value),
+    let node = AnyNodeRef::from(stmt);
+
+    if visitor.enter_node(node).is_traverse() {
```

Comment:
return early here to keep the indentation

Reply:
We need to call `leave_node` for all nodes, even if we're not traversing the node.

Reply:
Would this work?
```python
if !visitor.enter_node(node).is_traverse() {
    visitor.leave_node();
    return;
}
```
(or if you want more abstractions, `EnterNodeGuard` with a Drop impl)

Reply:
That would work but seemed more complicated than adding the extra indentation.